packages/mongo-nav/src/mongo-select/MongoSelect.tsx

```diff
@@ -89,140 +92,138 @@ const onKeyDown: React.KeyboardEventHandler = e => {
+          placeholder={current?.orgName ?? 'All Organizations'}
+          className={className}
+          urls={urls}
+          isActive={isActive}
+        />
+      }
+      className={menuContainerStyle}
+      justify="start"
+    >
+      <FocusableMenuItem>
+        <Input
+          onChange={onChange}
+          onKeyDown={onKeyDown}
+          variant="organization"
+        />
+      </FocusableMenuItem>
+
+      <>
+        {data?.map(renderOrganizationOption) ?? (
+          <li>
+            You do not belong to any organizations. Create an organization to
+            start using MongoDB Cloud
+          </li>
+        )}
+      </>
+
+      <MenuSeparator />
+      <MenuItem
+        onKeyDown={onKeyDown}
+        href={urls.mongoSelect?.viewAllOrganizations}
+      >
+        <strong className={viewAllStyle}>View All Organizations</strong>
+      </MenuItem>
+    </Menu>
+  );
+}
+
+OrgSelect.displayName = 'OrgSelect';
+
+export { OrgSelect };
+
+function ProjectSelect({
+  current,
+  className,
+  onChange,
+  data,
+  onClick,
+  constructProjectURL,
+  urls,
+}: ProjectMongoSelectProps) {
+  const renderProjectOption = (datum: ProjectInterface) => {
+    const { projectId, projectName, orgId } = datum;
 
     return (
       <MenuItem
-        key={datum.orgId}
-        className={menuItemContainerStyle}
+        key={projectId}
+        className={cx(menuItemContainerStyle, nameStyle)}
         onClick={onClick}
+        href={constructProjectURL(orgId, projectId)}
       >
-        {content}
+        {projectName}
       </MenuItem>
     );
   };
 
   return (
-    <Menu trigger={trigger} className={menuContainerStyle} justify="start">
+    <Menu
+      trigger={
+        <ProjectTrigger
+          placeholder={current.projectName}
+          className={className}
+        />
+      }
+      className={menuContainerStyle}
+      justify="start"
+    >
       <FocusableMenuItem>
-        <Input onChange={onChange} onKeyDown={onKeyDown} variant={variant} />
+        <Input onChange={onChange} onKeyDown={onKeyDown} variant="project" />
       </FocusableMenuItem>
 
-      <li role="none" className={listContainerStyle}>
-        <ul
-          onKeyDown={e => e.preventDefault()}
-          className={ulContainerStyle}
-          role="menu"
-        >
-          {isProject(props)
-            ? props.data.map(datum => renderProjectOption(props, datum))
-            : props.data.map(datum => renderOrganizationOption(props, datum))}
-        </ul>
-      </li>
+      <>{data?.map(datum => renderProjectOption(datum))}</>
 
       <MenuSeparator />
 
-      {footer}
+      <li onKeyDown={onKeyDown} role="none" className={projectButtonStyle}>
+        <FocusableMenuItem>
+          <Button href={urls.mongoSelect.viewAllProjects as string}>
+            View All Projects
+          </Button>
+        </FocusableMenuItem>
+        <FocusableMenuItem>
+          <Button href={urls.mongoSelect.newProject as string}>
```

Comment:
This typecasting is throwing red flags for me. We shouldn't need to do this, and if it's complaining, we're probably doing something that's not type-safe or typed correctly.

Reply:
<Required> only addresses the top-level, I was trying to be DRY but it seems like I need to make a new URLSInterface type that has every level <Required>